test/unit/rules/no-obscure-array-access-test.js

```diff
@@ -0,0 +1,65 @@
+import { ERROR_MESSAGE } from '../../../lib/rules/no-obscure-array-access.js';
+import generateRuleTests from '../../helpers/rule-test-harness.js';
+
+generateRuleTests({
+  name: 'no-obscure-array-access',
+
+  config: true,
+
+  good: [
+    "{{foo bar=(get this.list '0' )}}",
+    "<Foo @bar={{get this.list '0'}}",
+    "{{get this.list '0'}}",
+    '{{foo bar @list}}',
+    'Just a regular text in the template bar.[1] bar.1',
+    '<Foo foo="bar.[1]" />',
+  ],
+
+  bad: [
+    {
+      template: '{{foo bar=this.list.[0]}}',
+      result: {
```

Comment:
Can you update this to use `verifyResults` snapshot testing like the other rule tests. We prefer that now.